components/nocodb/nocodb.app.mjs

```diff
@@ -1,11 +1,147 @@
+import { Api } from "nocodb-sdk";
+
 export default {
   type: "app",
   app: "nocodb",
-  propDefinitions: {},
+  propDefinitions: {
+    projectId: {
+      type: "string",
+      label: "Project Id",
+      description: "The id of the project",
+      async options() {
+        const { list } = await this.listProjects();
+        return list.map((project) => ({
+          label: project.title,
+          value: project.id,
+        }));
+      },
+    },
+    tableName: {
+      type: "string",
+      label: "Table Name",
+      description: "The name of the table",
+      withLabel: true,
+      async options({ projectId }) {
+        const { list } = await this.listTables({
+          projectId,
+        });
+        return list.map((table) => ({
+          label: table.title,
+          value: table.id,
+        }));
+      },
+    },
+    rowId: {
+      type: "string",
+      label: "Row Id",
+      description: "Id of the row",
+    },
+    data: {
+      type: "any",
+      label: "data",
+      description: "Data that will be recorded",
+    },
+    fields: {
```

Comment:
Would there be a way to list the fields for a row type?

Reply:
there is no endpoint to get that, but I can get a row to create fields, what do you think?